File: frontend/quiz/src/seleniumTests/Pages/DriverFactory.js
import { Builder } from "selenium-webdriver";
import LogInPage from "./LogInPage";
class DriverFactory {
  async createDriverChrome() {
    let builder = new Builder()
      .usingServer()
      .withCapabilities({ browserName: "chrome" });
    return builder.build();
  }

  async createAndOpenChrome() {
    let driver = await this.createDriverChrome();
    const logInPage = new LogInPage(driver);
    await logInPage.open();
    return driver;
  }
}

export default DriverFactory;

File: backend/selenium/driverFactory.py
from selenium import webdriver
# import LogInPage from "./LogInPage";


class DriverFactory():

    def createDriverChrome(self):
        driver = webdriver.Chrome()
        driver.implicitly_wait(10)
        return driver

    def createAndOpenChrome(self):
        driver = webdriver.Chrome()
        driver.implicitly_wait(10)
        driver.get('http://localhost:3000/')
        driver.implicitly_wait(10)
        return driver
